### PR TITLE
[SPARK-4943][SPARK-5251][SQL] Allow table name having dot for db/catalog

### PR DESCRIPTION
**Pull request by scwf**
Follow up for #3941 and #3960, in this PR 
1 enable ddl table name having dot
2 enable hivecontext related api support dot table name with `tableIdentifier`
3 remove `QualifiedTableName` which do not match the other API, using tableIdentifier instead.

### CODE REVIEW

File: sql/hive/src/main/scala/org/apache/spark/sql/hive/HiveContext.scala [hunk 1]
```diff
@@ -95,12 +95,12 @@ class HiveContext(sc: SparkContext) extends SQLContext(sc) {
...
   def refreshTable(tableName: String): Unit = {
     // TODO: Database support...
-    catalog.refreshTable("default", tableName)
+    catalog.refreshTable(Seq("default", tableName))
   }
 
   protected[hive] def invalidateTable(tableName: String): Unit = {
     // TODO: Database support...
-    catalog.invalidateTable("default", tableName)
+    catalog.invalidateTable(Seq("default", tableName))
```
Comment (marmbrus) [inline sql/hive/src/main/scala/org/apache/spark/sql/hive/HiveContext.scala]: Why hardcode default here?  When a database isn't specified its up to the catalog to use the current one, where as this will ignore any `USE` statements that have been run.

Comment (scwf) [inline sql/hive/src/main/scala/org/apache/spark/sql/hive/HiveContext.scala]: yes, this just inherited from the old version. Now ```invalidateTable``` do not support database as you suggest, so here is a todo in line 102, how about make another PR to fix this? or do it in this PR

Comment (scwf) [inline sql/hive/src/main/scala/org/apache/spark/sql/hive/HiveContext.scala]: @marmbrus I will make a new PR to fix the database issue after this in.

**Comment by scwf** [comment]
ping @marmbrus

**Comment by scwf** [comment]
ping @marmbrus, is this ok to go?

**Comment by scwf** [comment]
ping

**Comment by marmbrus** [comment]
Okay, sorry this didn't make it into 1.3.  Mind bringing it up to date?

**Comment by scwf** [comment]
yeah later back to update this

**Comment by liancheng** [comment]
Sorry... Would you mind to rebase this again? Thanks!

**Comment by scwf** [comment]
Updated and added test cases.

**Comment by scwf** [comment]
To summarize this, in this PR: 
1 enable ddl table name having dot
2 enable HiveContext/DataFrame/SQLContext related api support dot table name with tableIdentifier
3 remove QualifiedTableName which do not match the other API from HiveMetastoreCatalog,  using tableIdentifier instead.
4 tablename x.y.z.....m.n indicate that m is the db name and z is the table name

File: sql/hive/src/main/scala/org/apache/spark/sql/hive/HiveMetastoreCatalog.scala [hunk 2]
```diff
@@ -173,19 +175,15 @@ private[hive] class HiveMetastoreCatalog(hive: HiveContext) extends Catalog with
     }
   }
 
-  def hiveDefaultTableFilePath(tableName: String): String = synchronized {
-    val currentDatabase = client.getDatabase(hive.sessionState.getCurrentDatabase)
+  def hiveTableFilePath(tableName: String): String = synchronized {
```
Comment (scwf) [inline sql/hive/src/main/scala/org/apache/spark/sql/hive/HiveMetastoreCatalog.scala]: should not rename this method

**Comment by scwf** [comment]
Jenkins, retest this please

**Comment by scwf** [comment]
@liancheng ,  rebased and updated, any comments here?

**Comment by scwf** [comment]
Jenkins, retest this please

**Comment by scwf** [comment]
ping
